exonum/src/storage/proof_list_index/tests.rs

```diff
@@ -838,32 +831,43 @@ mod list_hash_tests {
     use hex::FromHex;
 
     use crypto::Hash;
-    use storage::{hash::list_hash, Database, MemoryDB, ProofListIndex};
+    use storage::{hash::HashTag, Database, ListProof, MemoryDB, ProofListIndex};
 
     #[test]
-    fn proof_of_absence() {
+    fn proof_of_absence_single() {
         let db = MemoryDB::new();
         let mut fork = db.fork();
-        let mut index = ProofListIndex::new("absence", &mut fork);
-
-        index.push(vec![1]);
-        index.push(vec![2]);
-        index.push(vec![3]);
-        index.push(vec![4]);
-        index.push(vec![5]);
+        let mut list = ProofListIndex::new("absence", &mut fork);
 
-        let actual_list_hash = index.list_hash();
+        list.push(vec![1]);
+        list.push(vec![2]);
+        list.push(vec![3]);
+        list.push(vec![4]);
+        list.push(vec![5]);
 
-        let existed_index = 5u64;
         let root_hash =
             Hash::from_hex("5ba859b4d1799cb27ece9db8f7a76a50fc713a5d9d22f753eca42172996a88f9")
                 .unwrap();
 
-        let hash = list_hash(existed_index, root_hash);
-        assert_eq!(hash, actual_list_hash);
-
         let non_existed_index = 6u64;
-        let hash = list_hash(non_existed_index, index.merkle_root());
-        assert_ne!(hash, actual_list_hash);
+        let proof = list.get_proof(non_existed_index);
+
+        match proof {
+            ListProof::Absent(index, actual_list_hash) => {
+                assert_eq!(index, non_existed_index);
+
+                let expected_hash = HashTag::list_hash(list.len(), root_hash);
+                assert_eq!(expected_hash, actual_list_hash);
+                assert!(index >= list.len());
+            }
+            _ => {
+                panic!("Unexpected proof {:?}", proof);
+            }
+        }
+    }
+
+    #[test]
+    fn proof_of_absence_range() {
+        //TBD
```

Comment:
I'd suggest using something like this for marking tests that need to be implemented.

```rust
#[test]
#[ignore]
fn proof_of_absence_range() {
    unimplemented!()
}
```

This way they are more conspicuous in test runner output.